### PR TITLE
feat(api-client,core,commons): Chunked /users requests & types ArrayUtils

### PR DESCRIPTION
**Pull request by Yserz**

### CODE REVIEW

File: packages/api-client/src/main/user/UserAPI.ts [hunk 1]
```diff
@@ -38,6 +39,7 @@ import {
 } from '../user/';
 
 export class UserAPI {
+  static DEFAULT_USERS_CHUNK_SIZE = 50;
```
Comment (ffflorian) [inline packages/api-client/src/main/user/UserAPI.ts]:
```suggestion
  static readonly DEFAULT_USERS_CHUNK_SIZE = 50;
```

Or will we change this value?

File: packages/api-client/src/main/user/UserAPI.ts [hunk 2]
```diff
@@ -254,7 +256,28 @@ export class UserAPI {
    * @param parameters Multiple user's handles or IDs
    * @see https://staging-nginz-https.zinfra.io/swagger-ui/#!/users/users
    */
-  public getUsers(parameters: {handles?: string[]; ids?: string[]}): Promise<User[]> {
+  public async getUsers(
+    parameters: {handles?: string[]; ids?: string[]},
```
Comment (ffflorian) [inline packages/api-client/src/main/user/UserAPI.ts]:
```suggestion
    parameters: {handles?: string[]; ids?: string[]} = {},
```

File: packages/api-client/src/main/user/UserAPI.ts [hunk 3]
```diff
@@ -254,7 +256,28 @@ export class UserAPI {
...
+      const uniqueHandles = removeDuplicates(handles);
+      const handleChunks = chunk(uniqueHandles, limit);
+      const tasks = handleChunks.map(handleChunk => this._getUsers({handles: handleChunk}));
+      return Promise.all(tasks).then(flatten);
+    } else if (ids && ids.length) {
+      const uniqueIds = removeDuplicates(ids);
+      const idChunks = chunk(uniqueIds, limit);
+      const tasks = idChunks.map(idChunk => this._getUsers({ids: idChunk}));
+      return Promise.all(tasks).then(flatten);
+    } else {
```
Comment (ffflorian) [inline packages/api-client/src/main/user/UserAPI.ts]: `else` is not needed

File: packages/api-client/src/main/user/UserAPI.ts (hunk 2 above)
Comment (Yserz) [reply packages/api-client/src/main/user/UserAPI.ts]: It's exclusive so it is not optional

Comment (ffflorian) [reply packages/api-client/src/main/user/UserAPI.ts]: Then we should mark this as exclusive with function overloading.

File: packages/api-client/src/main/user/UserAPI.ts [hunk 4]
```diff
@@ -254,7 +256,29 @@ export class UserAPI {
...
+    limit: number = UserAPI.DEFAULT_USERS_CHUNK_SIZE
+  ): Promise<User[]> {
+    const {handles, ids} = parameters;
+
+    if (handles && handles.length) {
+      const uniqueHandles = removeDuplicates(handles);
+      const handleChunks = chunk(uniqueHandles, limit);
+      const tasks = handleChunks.map(handleChunk => this._getUsers({handles: handleChunk}));
+      return Promise.all(tasks).then(flatten);
+    } else if (ids && ids.length) {
```
Comment (ffflorian) [inline packages/api-client/src/main/user/UserAPI.ts]:
```suggestion
    }
    
    if (ids && ids.length) {
```

File: packages/api-client/src/main/user/UserAPI.ts [hunk 5]
```diff
@@ -254,7 +256,29 @@ export class UserAPI {
...
+      const uniqueIds = removeDuplicates(ids);
+      const idChunks = chunk(uniqueIds, limit);
+      const tasks = idChunks.map(idChunk => this._getUsers({ids: idChunk}));
+      return Promise.all(tasks).then(flatten);
+    }
+    return [];
```
Comment (ffflorian) [inline packages/api-client/src/main/user/UserAPI.ts]:
```suggestion

    return [];
```

File: packages/api-client/src/main/user/UserAPI.ts [hunk 6]
```diff
@@ -254,7 +256,29 @@ export class UserAPI {
    * @param parameters Multiple user's handles or IDs
    * @see https://staging-nginz-https.zinfra.io/swagger-ui/#!/users/users
    */
-  public getUsers(parameters: {handles?: string[]; ids?: string[]}): Promise<User[]> {
+  public async getUsers(parameters: {ids: string[]}, limit?: number): Promise<User[]>;
+  public async getUsers(parameters: {handles: string[]}, limit?: number): Promise<User[]>;
+  public async getUsers(
+    parameters: {ids?: string[]; handles?: string[]},
+    limit: number = UserAPI.DEFAULT_USERS_CHUNK_SIZE
```
Comment (ffflorian) [inline packages/api-client/src/main/user/UserAPI.ts]: 👍

**Pull request merged**